cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -60,13 +60,28 @@ func CreateOrUpdateConfigMap(client clientset.Interface, cm *v1.ConfigMap) error
 // to conflicts, and a retry will be issued if the ConfigMap was modified on the server between the refresh and the update (while the mutation was
 // taking place)
 func CreateOrMutateConfigMap(client clientset.Interface, cm *v1.ConfigMap, mutator ConfigMapMutator) error {
-	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(cm); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			return errors.Wrap(err, "unable to create ConfigMap")
+	var lastError error
+	err := wait.ExponentialBackoff(wait.Backoff{
+		Steps:    20,
+		Duration: 500 * time.Millisecond,
+		Factor:   1.0,
+		Jitter:   0.1,
+	}, func() (bool, error) {
+		if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(cm); err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				err := MutateConfigMap(client, metav1.ObjectMeta{Namespace: cm.ObjectMeta.Namespace, Name: cm.ObjectMeta.Name}, mutator)
+				lastError = err
+				return err == nil, nil
+			}
+			lastError = err
```

Comment:
this can be:
```
if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(cm); err != nil {
    lastError = err
    if apierrors.IsAlreadyExists(err) {
    ...
}
```

Reply:
yes.

Reply:
ah sorry no, it's not the same. They are different errors.

Reply:
right, my mistake.